[BUG] Cannot build project from a sub-directory

**Describe the bug**
I wanted to just run a single test without building anything/etc. The IDE has issues without having to blow away the existing IDE setup for the plugin and start over, so I tried to do this with maven. The typical way to do this in maven is to go to the sub-directory of a project and build/test from there.  This does not work.

```
$ cd tests
$ env -u SPARK_HOME -u SPARK_CONF_DIR mvn clean install -Dpytest.TEST_TAGS="" -Dpytest.TEST_PARALLEL=9 -Dcuda.version=cuda11 -Dbuildver=320 -DwildcardSuites=com.nvidia.spark.rapids.AdaptiveQueryExecSuite
...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.408 s
[INFO] Finished at: 2021-09-13T05:54:57-05:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:1.8:run (generate-build-info) on project rapids-4-spark-tests_2.12: An Ant BuildException has occured: exec returned: 127
[ERROR] around Ant part ...<exec output=".../tests/target/extra-resources/rapids4spark-version-info.properties" failonerror="true" executable="bash">... @ 6:161 in .../tests/target/antrun/build-main.xml
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

I looked in CONTRIBUTING.md and there is nothing in there about this.

There are alternative ways to make this work, hint add `-pl tests`. I would prefer to have us fix this so it actually works, but I am fine with documenting the work around.  And even more so I would love it if we had a good way to make the IDE actually switch back and forth easily because `wildcardSuites` is not perfect.

--- a/integration_tests/pom.xml
+++ b/integration_tests/pom.xml
@@ -214,7 +214,6 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-antrun-plugin</artifactId>
-                <version>1.8</version>
                 <executions>
                     <execution>
                         <id>generate-build-info</id>
@@ -225,7 +224,7 @@
                                 <mkdir dir="${project.build.directory}/extra-resources"/>
                                 <mkdir dir="${project.build.directory}/tmp"/>
                                 <exec executable="bash" failonerror="true" output="${project.build.directory}/extra-resources/rapids4spark-version-info.properties">
-                                    <arg value="${user.dir}/build/build-info"/>
+                                    <arg value="${spark.rapids.source.basedir}/build/build-info"/>
                                     <arg value="${project.version}"/>
                                     <arg value="${cudf.version}"/>
                                 </exec>

--- a/pom.xml
+++ b/pom.xml
@@ -1022,7 +1022,7 @@
                                 <mkdir dir="${project.build.directory}/extra-resources"/>
                                 <mkdir dir="${project.build.directory}/tmp"/>
                                 <exec executable="bash" failonerror="true" output="${project.build.directory}/extra-resources/rapids4spark-version-info.properties">
-                                    <arg value="${user.dir}/build/build-info"/>
+                                    <arg value="${spark.rapids.source.basedir}/build/build-info"/>
                                     <arg value="${project.version}"/>
                                     <arg value="${cudf.version}"/>
                                 </exec>
@@ -1225,6 +1225,26 @@
                     </excludes>
                 </configuration>
             </plugin>
+
+            <!--use this plugin to configure "spark.rapids.source.basedir" property-->
+            <plugin>
+                <groupId>org.commonjava.maven.plugins</groupId>
+                <artifactId>directory-maven-plugin</artifactId>
+                <version>0.1</version>
+                <executions>
+                    <execution>
+                        <id>directories</id>
+                        <goals>
+                            <goal>highest-basedir</goal>
+                        </goals>
+                        <phase>initialize</phase>
+                        <configuration>
+                            <property>spark.rapids.source.basedir</property>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+
             <plugin>
                 <!--
                 This is an alternative implementation of the scalastyle check invocation,

--- a/shims/spark301/pom.xml
+++ b/shims/spark301/pom.xml
@@ -48,7 +48,7 @@
                             <target>
                                 <mkdir dir="${project.build.directory}/extra-resources"/>
                                 <exec executable="bash" failonerror="true" output="${project.build.directory}/extra-resources/spark-${spark301.version}-info.properties">
-                                    <arg value="${user.dir}/build/dependency-info.sh"/>
+                                    <arg value="${spark.rapids.source.basedir}/build/dependency-info.sh"/>
                                     <arg value="${cudf.version}"/>
                                     <arg value="${cuda.version}"/>
                                     <arg value="${spark301.version}"/>

--- a/shims/spark301db/pom.xml
+++ b/shims/spark301db/pom.xml
@@ -48,7 +48,7 @@
                             <target>
                                 <mkdir dir="${project.build.directory}/extra-resources"/>
                                 <exec executable="bash" failonerror="true" output="${project.build.directory}/extra-resources/spark-${spark301db.version}-info.properties">
-                                    <arg value="${user.dir}/build/dependency-info.sh"/>
+                                    <arg value="${spark.rapids.source.basedir}/build/dependency-info.sh"/>
                                     <arg value="${cudf.version}"/>
                                     <arg value="${cuda.version}"/>
                                     <arg value="${spark301db.version}"/>

--- a/shims/spark302/pom.xml
+++ b/shims/spark302/pom.xml
@@ -48,7 +48,7 @@
                             <target>
                                 <mkdir dir="${project.build.directory}/extra-resources"/>
                                 <exec executable="bash" failonerror="true" output="${project.build.directory}/extra-resources/spark-${spark302.version}-info.properties">
-                                    <arg value="${user.dir}/build/dependency-info.sh"/>
+                                    <arg value="${spark.rapids.source.basedir}/build/dependency-info.sh"/>
                                     <arg value="${cudf.version}"/>
                                     <arg value="${cuda.version}"/>
                                     <arg value="${spark302.version}"/>

--- a/shims/spark303/pom.xml
+++ b/shims/spark303/pom.xml
@@ -48,7 +48,7 @@
                             <target>
                                 <mkdir dir="${project.build.directory}/extra-resources"/>
                                 <exec executable="bash" failonerror="true" output="${project.build.directory}/extra-resources/spark-${spark303.version}-info.properties">
-                                    <arg value="${user.dir}/build/dependency-info.sh"/>
+                                    <arg value="${spark.rapids.source.basedir}/build/dependency-info.sh"/>
                                     <arg value="${cudf.version}"/>
                                     <arg value="${cuda.version}"/>
                                     <arg value="${spark303.version}"/>

--- a/shims/spark304/pom.xml
+++ b/shims/spark304/pom.xml
@@ -48,7 +48,7 @@
                             <target>
                                 <mkdir dir="${project.build.directory}/extra-resources"/>
                                 <exec executable="bash" failonerror="true" output="${project.build.directory}/extra-resources/spark-${spark304.version}-info.properties">
-                                    <arg value="${user.dir}/build/dependency-info.sh"/>
+                                    <arg value="${spark.rapids.source.basedir}/build/dependency-info.sh"/>
                                     <arg value="${cudf.version}"/>
                                     <arg value="${cuda.version}"/>
                                     <arg value="${spark304.version}"/>

--- a/shims/spark311/pom.xml
+++ b/shims/spark311/pom.xml
@@ -48,7 +48,7 @@
                             <target>
                                 <mkdir dir="${project.build.directory}/extra-resources"/>
                                 <exec executable="bash" failonerror="true" output="${project.build.directory}/extra-resources/spark-${spark311.version}-info.properties">
-                                    <arg value="${user.dir}/build/dependency-info.sh"/>
+                                    <arg value="${spark.rapids.source.basedir}/build/dependency-info.sh"/>
                                     <arg value="${cudf.version}"/>
                                     <arg value="${cuda.version}"/>
                                     <arg value="${spark311.version}"/>

--- a/shims/spark311cdh/pom.xml
+++ b/shims/spark311cdh/pom.xml
@@ -48,7 +48,7 @@
                             <target>
                                 <mkdir dir="${project.build.directory}/extra-resources"/>
                                 <exec executable="bash" failonerror="true" output="${project.build.directory}/extra-resources/spark-${spark311cdh.version}-info.properties">
-                                    <arg value="${user.dir}/build/dependency-info.sh"/>
+                                    <arg value="${spark.rapids.source.basedir}/build/dependency-info.sh"/>
                                     <arg value="${cudf.version}"/>
                                     <arg value="${cuda.version}"/>
                                     <arg value="${spark311cdh.version}"/>

--- a/shims/spark311db/pom.xml
+++ b/shims/spark311db/pom.xml
@@ -48,7 +48,7 @@
                             <target>
                                 <mkdir dir="${project.build.directory}/extra-resources"/>
                                 <exec executable="bash" failonerror="true" output="${project.build.directory}/extra-resources/spark-${spark311db.version}-info.properties">
-                                    <arg value="${user.dir}/build/dependency-info.sh"/>
+                                    <arg value="${spark.rapids.source.basedir}/build/dependency-info.sh"/>
                                     <arg value="${cudf.version}"/>
                                     <arg value="${cuda.version}"/>
                                     <arg value="${spark311db.version}"/>

--- a/shims/spark312/pom.xml
+++ b/shims/spark312/pom.xml
@@ -48,7 +48,7 @@
                             <target>
                                 <mkdir dir="${project.build.directory}/extra-resources"/>
                                 <exec executable="bash" failonerror="true" output="${project.build.directory}/extra-resources/spark-${spark312.version}-info.properties">
-                                    <arg value="${user.dir}/build/dependency-info.sh"/>
+                                    <arg value="${spark.rapids.source.basedir}/build/dependency-info.sh"/>
                                     <arg value="${cudf.version}"/>
                                     <arg value="${cuda.version}"/>
                                     <arg value="${spark312.version}"/>

--- a/shims/spark312db/pom.xml
+++ b/shims/spark312db/pom.xml
@@ -49,7 +49,7 @@
                                 <mkdir dir="${project.build.directory}/extra-resources"/>
                                 <exec executable="bash" failonerror="true"
                                       output="${project.build.directory}/extra-resources/spark-${spark312db.version}-info.properties">
-                                    <arg value="${user.dir}/build/dependency-info.sh"/>
+                                    <arg value="${spark.rapids.source.basedir}/build/dependency-info.sh"/>
                                     <arg value="${cudf.version}"/>
                                     <arg value="${cuda.version}"/>
                                     <arg value="${spark312db.version}"/>

--- a/shims/spark313/pom.xml
+++ b/shims/spark313/pom.xml
@@ -48,7 +48,7 @@
                             <target>
                                 <mkdir dir="${project.build.directory}/extra-resources"/>
                                 <exec executable="bash" failonerror="true" output="${project.build.directory}/extra-resources/spark-${spark313.version}-info.properties">
-                                    <arg value="${user.dir}/build/dependency-info.sh"/>
+                                    <arg value="${spark.rapids.source.basedir}/build/dependency-info.sh"/>
                                     <arg value="${cudf.version}"/>
                                     <arg value="${cuda.version}"/>
                                     <arg value="${spark313.version}"/>

--- a/shims/spark320/pom.xml
+++ b/shims/spark320/pom.xml
@@ -49,7 +49,7 @@
                 <mkdir dir="${project.build.directory}/extra-resources"/>
                 <exec executable="bash" failonerror="true"
                       output="${project.build.directory}/extra-resources/spark-${spark320.version}-info.properties">
-                  <arg value="${user.dir}/build/dependency-info.sh"/>
+                  <arg value="${spark.rapids.source.basedir}/build/dependency-info.sh"/>
                   <arg value="${cudf.version}"/>
                   <arg value="${cuda.version}"/>
                   <arg value="${spark320.version}"/>

--- a/shims/spark321/pom.xml
+++ b/shims/spark321/pom.xml
@@ -49,7 +49,7 @@
                 <mkdir dir="${project.build.directory}/extra-resources"/>
                 <exec executable="bash" failonerror="true"
                       output="${project.build.directory}/extra-resources/spark-${spark321.version}-info.properties">
-                  <arg value="${user.dir}/build/dependency-info.sh"/>
+                  <arg value="${spark.rapids.source.basedir}/build/dependency-info.sh"/>
                   <arg value="${cudf.version}"/>
                   <arg value="${cuda.version}"/>
                   <arg value="${spark321.version}"/>

--- a/shims/spark330/pom.xml
+++ b/shims/spark330/pom.xml
@@ -49,7 +49,7 @@
                 <mkdir dir="${project.build.directory}/extra-resources"/>
                 <exec executable="bash" failonerror="true"
                       output="${project.build.directory}/extra-resources/spark-${spark330.version}-info.properties">
-                  <arg value="${user.dir}/build/dependency-info.sh"/>
+                  <arg value="${spark.rapids.source.basedir}/build/dependency-info.sh"/>
                   <arg value="${cudf.version}"/>
                   <arg value="${cuda.version}"/>
                   <arg value="${spark330.version}"/>

